python/lsst/jointcal/jointcal.py

```diff
@@ -772,35 +774,79 @@ def _check_stars(self, associations):
+            The model to minimize with ``fitter``.
+        max_steps : `int`
+            Maximum number of steps to run outlier rejection before declaring
+            convergence failure.
+        name : {'photometry' or 'astrometry'}
+            What type of data are we fitting (for logs and debugging files).
+        whatToFit : `str`
+            Passed to ``fitter.minimize()`` to define the parameters to fit.
+        dataName : str, optional
+            Descriptive name for this dataset (e.g. tract and filter),
+            for debugging.
+        doRankUpdate : bool, optional
+            Do an Eigen rank update during minimization, or recompute the full
+            matrix and gradient?
+        doLineSearch : bool, optional
+            Do a line search for the optimum step during minimization?
+
+        Returns
+        -------
+        chi2: `lsst.jointcal.Chi2Statistic`
+            The final chi2 after the fit converges, or is forced to end.
+
+        Raises
+        ------
+        FloatingPointError
+            Raised if the fitter fails with a non-finite value.
+        RuntimeError
+            Raised if the fitter fails for some other reason;
+            log messages will provide further details.
+        """
 
         dumpMatrixFile = "%s_postinit" % name if self.config.writeInitMatrix else ""
         for i in range(max_steps):
             # outlier removal at 5 sigma.
-            r = fit.minimize(whatToFit,
-                             self.config.outlierRejectSigma,
-                             doRankUpdate=doRankUpdate,
-                             doLineSearch=doLineSearch,
-                             dumpMatrixFile=dumpMatrixFile)
+            r = fitter.minimize(whatToFit,
+                                self.config.outlierRejectSigma,
+                                doRankUpdate=doRankUpdate,
+                                doLineSearch=doLineSearch,
+                                dumpMatrixFile=dumpMatrixFile)
             dumpMatrixFile = ""  # clear it so we don't write the matrix again.
-            chi2 = fit.computeChi2()
+            chi2 = fitter.computeChi2()
             self._check_stars(associations)
-            if not np.isfinite(chi2.chi2):
-                raise FloatingPointError('Fit iteration chi2 is invalid: %s'%chi2)
             self.log.info(str(chi2))
             if r == MinimizeResult.Converged:
                 if doRankUpdate:
                     self.log.debug("fit has converged - no more outliers - redo minimization "
                                    "one more time in case we have lost accuracy in rank update.")
                     # Redo minimization one more time in case we have lost accuracy in rank update
-                    r = fit.minimize(whatToFit, 5)  # outliers removal at 5 sigma.
-                chi2 = fit.computeChi2()
+                    r = fitter.minimize(whatToFit, 5)  # outliers removal at 5 sigma.
```

Comment:
Is this really meant to be hardcoded to 5? I would have expected it to use self.config.outlierRejectSigma, otherwise that should be named something like outlierRejectSigmaInitial.